src/colorizer/ColorizeCanvas.ts

```diff
@@ -75,6 +75,7 @@ const getDefaultUniforms = (): ColorizeUniforms => {
   };
 };
 
+// TODO: Change into a component?
```

Comment:
Some rambling about React component conversion: I'd advocate for keeping classes within this directory relatively unchanged rather than allowing React to penetrate too deeply into the inner workings of the app, though I'm not super committed to this preference. We are at least required to draw a line between what gets handled in React's functional paradigm vs. what React hands off via effects to be resolved imperatively, and I think we should be deliberate about where we put that line. Since this class's main purpose is to own a `canvas`, an element which must be manipulated imperatively via its methods, I think the boundaries of this class are a pretty natural place for that boundary to go.

I'm less certain about `RecordingControls`, `TimeControls`, `ColorRamp`, etc., though I will say: we have the option to wrap these classes in components rather than converting them to components. This would provide all the same ergonomic benefits but might provide greater separation of concerns.

Reply:
For context on that comment line, I think there was one main reason to think about making this a component, and that was to drive redraws using props passed in from its parent. (Aka The reworking of the drawLoop stuff, for which there are other valid ways to handle it..).